umad/src/main/java/com/pkukielka/AccessMonitorRewriter.java

```diff
@@ -11,61 +13,62 @@
     long timestamp;
     long threadId;
     int hashCode;
-    String stackTrace;
+    StackTraceElement[] stackTrace;
     String threadName;
+    Set<Long> locks;
 
-    LastAccess(long timestamp, long threadId, int hashCode, String threadName) {
+    LastAccess(long timestamp, long threadId, int hashCode, String threadName, Set<Long> locks) {
         this.timestamp = timestamp;
         this.threadId = threadId;
         this.hashCode = hashCode;
         this.stackTrace = null;
         this.threadName = threadName;
+        this.locks = locks;
     }
 }
 
 public class AccessMonitorRewriter extends MethodRewriter {
     private static final AppConfig conf = AppConfig.get();
-    private static final Map<String, LastAccess> methodCalls = new HashMap<String, LastAccess>();
-    private static final Set<String> alreadyReported = new HashSet<String>();
+    private static final Map<String, LastAccess> methodCalls = new HashMap<>();
+    private static final Set<String> alreadyReported = new HashSet<>();
+    private static String locks;
 
-    private final Set<String> safeMethodTypes = new HashSet<String>();
-    private final Set<String> safeMethods = new HashSet<String>();
-    private final Set<String> safeFieldTypes = new HashSet<String>();
-
-    private static final Set<String> synchronizeIndicators = new HashSet<String>();
-
-    private static final int STACK_TRACE_LENGTH = 5;
+    private static final int STACK_TRACE_LENGTH = 10;
 
     private static final int realStackStartIndex = 2;
 
-    private static int safetyChecksRecursionLevel = 1;
+    AccessMonitorRewriter(Config config) {
+        super(config);
+        locks = config.getString("locks");
```

Comment:
It should be called `getLocksCode` or anything that suggest that this is java code.